server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -161,11 +161,11 @@ public EnvironmentConfig getEnvironmentConfig(String environmentName) throws NoS
         return environments.named(new CaseInsensitiveString(environmentName));
     }
 
-    public EnvironmentConfig forEdit(String environmentName) {
-        return cloner.deepClone(environments.find(new CaseInsensitiveString(environmentName)));
+    public EnvironmentConfig getEnvironmentForEdit(String environmentName) {
```

Comment:
@GaneshSPatil - Can you write a test to reproduce the bug that you fixed. This specific change is the fix for the bug, but its not very evident as to what was the bug, so there is nothing stopping anyone from changing this back to `environments.find`